src/file.js

```diff
@@ -207,13 +229,25 @@ Sk.builtin.file.prototype["truncate"] = new Sk.builtin.func(function truncate(se
 Sk.builtin.file.prototype["write"] = new Sk.builtin.func(function write(self, str) {
     var mode = Sk.ffi.remapToJs(self.mode);
     if (mode === "w" || mode === "wb" || mode === "a" || mode === "ab") {
-        if (self.fileno === 1) {
-            Sk.output(Sk.ffi.remapToJs(str));
+        if (Sk.filewrite) {
+            if (self.closed) {
+                throw new Sk.builtin.ValueError("I/O operation on closed file");
+            }
+
+            if (self.fileno === 1) {
+                Sk.output(Sk.ffi.remapToJs(str));
+            } else {
+                Sk.filewrite(self, str);
```

Comment:
Since Sk.filewrite is configurable should we not remap str to JS instead of making the configured  external function handle it??

Reply:
Yeah I guess that makes sense.